src/nvim/testdir/test_exists.vim

```diff
@@ -70,6 +70,9 @@ func Test_exists()
   call assert_equal(1, exists('$EDITOR_NAME'))
   " Non-existing environment variable
   call assert_equal(0, exists('$NON_ENV_VAR'))
+  " Empty environment variable
+  let $EDITOR_NAME = ''
+  call assert_equal(1, exists('$EDITOR_NAME'))
```

Comment:
So Vim is unlikely to fix it, and maybe not even taking (a different) test here.
Better to have this as a busted test then I assume?